### PR TITLE
feature: adapt library to Discord.js v14

### PR DESCRIPTION
**Pull request by DrakeXorn**

### CODE REVIEW

File: src/SelfRoleManager.ts [hunk 1]
```diff
@@ -43,48 +46,53 @@ export class SelfRoleManager extends EventEmitter {
   /**
    * Creates an instance of SelfRoleManager.
    * @param {Client} [client] The client that instantiated this Manager
-   * @param {SelfRoleOptions} [options={
+   * @param {SelfRoleOptions} [roleOptions={
```
Comment (HunteRoi) [inline src/SelfRoleManager.ts]: The renaming of this parameter is not part of this PR nor should it be acceptable considering the constructor is part of the public API already available to the users of the library. Thus we should avoid possible breaking changes such as this.

File: src/SelfRoleManager.ts [hunk 2]
```diff
@@ -43,48 +46,53 @@ export class SelfRoleManager extends EventEmitter {
...
+    if (!intents.has(IntentsBitField.Flags.Guilds)) {
+      throw new Error(
+        'Guilds intent bitfield is required to use this package!'
+      );
     }
-    if (!intents.has(Intents.FLAGS.GUILD_MEMBERS)) {
-      throw new Error('GUILD_MEMBERS intent is required to use this package!');
+    if (!intents.has(IntentsBitField.Flags.GuildMembers)) {
+      throw new Error(
+        'GuildMembers intent bitfield is required to use this package!'
```
Comment (HunteRoi) [inline src/SelfRoleManager.ts]: Updating the error message would be a huge breaking change and in this case is not essential to the update to discord.js v14. I am totally against introducing such changes in a context of a inner package version upgrade.

**Review by HunteRoi — CHANGES_REQUESTED** [review]

File: src/SelfRoleManager.ts (hunk 2 above)
Comment (DrakeXorn) [reply src/SelfRoleManager.ts]: I understand your point of view... I thought I did well as the intent fields changed, but no problem, I put the messages back

Comment (HunteRoi) [reply src/SelfRoleManager.ts]: It is fine, I understand your position as you wish to keep the error message consistent with the condition. However, considering error messages part of the public API is important. We have the chance to not have named parameters in TypeScript/JavaScript but could you imagine in C# for example?

**Review by HunteRoi — APPROVED** [review]
